polygeist/test/polygeist-opt/sycl/kernel_disjoint_specialization.mlir

```diff
@@ -105,48 +97,34 @@ gpu.module @device_func {
   // CHECK-NEXT:    %c1 = arith.constant 1 : index
-  // CHECK-NEXT:    %c0_i32_3 = arith.constant 0 : i32
-  // CHECK-NEXT:    %4 = sycl.range.get %alloca_1[%c0_i32_3] : (memref<1x!sycl_range_2_>, i32) -> index
-  // CHECK-NEXT:    %5 = arith.subi %4, %c1 : index
-  // CHECK-NEXT:    %c1_i32_4 = arith.constant 1 : i32
-  // CHECK-NEXT:    %6 = sycl.range.get %alloca_1[%c1_i32_4] : (memref<1x!sycl_range_2_>, i32) -> index
-  // CHECK-NEXT:    %alloca_5 = memref.alloca() : memref<1x!sycl_id_2_>
-  // CHECK-NEXT:    %c0_6 = arith.constant 0 : index
-  // CHECK-NEXT:    %c0_i32_7 = arith.constant 0 : i32
-  // CHECK-NEXT:    %7 = sycl.id.get %alloca_5[%c0_i32_7] : (memref<1x!sycl_id_2_>, i32) -> memref<?xindex>
-  // CHECK-NEXT:    memref.store %5, %7[%c0_6] : memref<?xindex>
-  // CHECK-NEXT:    %c1_i32_8 = arith.constant 1 : i32
-  // CHECK-NEXT:    %8 = sycl.id.get %alloca_5[%c1_i32_8] : (memref<1x!sycl_id_2_>, i32) -> memref<?xindex>
-  // CHECK-NEXT:    memref.store %6, %8[%c0_6] : memref<?xindex>
-  // CHECK-NEXT:    [[ACC1_END:%.*]] = sycl.accessor.subscript %arg0[%alloca_5] : (memref<?x!sycl_accessor_2_f32_r_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
+  // CHECK-NEXT:    %c0_i32 = arith.constant 0 : i32
+  // CHECK-NEXT:    %3 = sycl.range.get %alloca[%c0_i32] : (memref<1x!sycl_range_2_>, i32) -> index
+  // CHECK-NEXT:    %4 = arith.subi %3, %c1 : index
+  // CHECK-NEXT:    %c1_i32 = arith.constant 1 : i32
+  // CHECK-NEXT:    %5 = sycl.range.get %alloca[%c1_i32] : (memref<1x!sycl_range_2_>, i32) -> index
+  // CHECK-NEXT:    %6 = sycl.id.constructor(%4, %5) : (index, index) -> memref<1x!sycl_id_2_>
+  // CHECK-NEXT:    [[ACC1_END:%.*]] = sycl.accessor.subscript %arg0[%6] : (memref<?x!sycl_accessor_2_f32_r_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
 
   // COM: Version with condition: [[ACC1_END]] <= [[ACC2_BEGIN]] || [[ACC1_BEGIN]] >= [[ACC2_END]].
-  // CHECK:         [[ACC2_BEGIN:%.*]] = sycl.accessor.subscript %arg1[%alloca_10] : (memref<?x!sycl_accessor_2_f32_w_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
-  // CHECK:         [[ACC2_END:%.*]] = sycl.accessor.subscript %arg1[%alloca_19] : (memref<?x!sycl_accessor_2_f32_w_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
+  // CHECK:         [[ACC2_BEGIN:%.*]] = sycl.accessor.subscript %arg1[%8] : (memref<?x!sycl_accessor_2_f32_w_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
+  // CHECK:         [[ACC2_END:%.*]] = sycl.accessor.subscript %arg1[%14] : (memref<?x!sycl_accessor_2_f32_w_gb>, memref<1x!sycl_id_2_>) -> memref<?xf32, 1>
   // CHECK-DAG:     [[ACC1_END_PTR:%.*]] = "polygeist.memref2pointer"([[ACC1_END]]) : (memref<?xf32, 1>) -> !llvm.ptr<1>
   // CHECK-DAG:     [[ACC2_BEGIN_PTR:%.*]]  = "polygeist.memref2pointer"([[ACC2_BEGIN]]) : (memref<?xf32, 1>) -> !llvm.ptr<1>
-  // CHECK-NEXT:    %22 = llvm.icmp "ule" [[ACC1_END_PTR]], [[ACC2_BEGIN_PTR]] : !llvm.ptr<1>
+  // CHECK-NEXT:    %18 = llvm.icmp "ule" [[ACC1_END_PTR]], [[ACC2_BEGIN_PTR]] : !llvm.ptr<1>
```

Comment:
Nit: Pattern-match values here (and in the other tests you're modifying).

Reply:
done.